tests/std/tests/VSO_0000000_vector_algorithms/test.cpp

```diff
@@ -1247,22 +1247,52 @@ void test_case_string_find_last_of(const basic_string<T>& input_haystack, const
     assert(expected == actual);
 }
 
+template <class T>
+void test_case_string_find_str(const basic_string<T>& input_haystack, const basic_string<T>& input_needle) {
+    ptrdiff_t expected;
+    if (input_needle.empty()) {
+        expected = 0;

```

Comment:
Apparently this special case is not needed for `find`. It may be needed for (future) `rfind` though.

Reply:
This is fine - usually I am against adding special cases when the main case handles it correctly, but establishing a pattern for later tests is a reasonable justification, and it's not a very disruptive special case either.